src/aspire/classification/averager2d.py

```diff
@@ -725,6 +750,211 @@ def average(
         return AligningAverager2D.average(self, classes, reflections, coefs)
 
 
+class BFTAverager2D(AligningAverager2D):
+    """
+    This perfoms a Brute Force Translations and fast rotational alignment.
+
+    For each shift,
+       Perform polar Fourier cross correlation based rotational alignment.
+
+    Return the rotation and shift yielding the best results.
+    """
+
+    def __init__(
+        self,
+        composite_basis,
+        src,
+        alignment_basis=None,
+        n_angles=360,
+        n_radial=None,
+        radius=None,
+        sub_pixel=0.1,
+        batch_size=512,
+        dtype=None,
+    ):
+        """
+        See AligningAverager2D. Adds `n_angles`, `n_radial`, `radius`, `sub_pixel`.
+
+        :params n_angles: Number of PFT angular components, defaults 360.
+        :param n_radial: Number of PFT radial components, defaults `self.src.L//2`.
+        :param radius: Brute force translation search radius.
+            `0` disables translation search, rotations only.
+            Defaults to `src.L//32`.
+        :param sub_pixel: Subpixel shift size used in brute force shift search.
+        """
+        super().__init__(
+            composite_basis,
+            src,
+            alignment_basis,
+            batch_size=batch_size,
+            dtype=dtype,
+        )
+
+        self.n_angles = n_angles
+
+        self.radius = radius if radius is not None else src.L // 32
+
+        if self.radius != 0:
+
+            if not hasattr(self.alignment_basis, "shift"):
+                raise RuntimeError(
+                    f"{self.__class__.__name__}'s alignment_basis {self.alignment_basis} must provide a `shift` method."
+                )
+
+        self.sub_pixel = sub_pixel
+
+        # Configure number of radial points
+        self.n_radial = n_radial or self.src.L // 2
+
+        # Setup Polar Transform
+        self._pft = PolarFT(
+            self.src.L, ntheta=n_angles, nrad=self.n_radial, dtype=self.dtype
+        )
+        self._mask = xp.asarray(grid_2d(self.src.L, normalized=True)["r"] < 1)
+
+    def _fast_rotational_alignment(self, pfA, pfB):
+        """
+        Perform fast rotational alignment using Polar Fourier cross correlation.
+
+        Note broadcasting is specialized for this problem.
+        pfA.shape (m, nt, nr)
```

Comment:
I'm guessing here `nt` is number of angles and `nr` is number of radial points?

Reply:
Yeah, just dev notes.  I updated string `ntheta` and `nrad`.  I can move to code comment if that is better.